src/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD114AvoidReturningNullTaskCodeFixTests.cs

```diff
@@ -177,5 +177,143 @@ class Test
 
             await Verify.VerifyCodeFixAsync(test, withFix);
         }
+
+        [Fact(Skip = "Cannot find generic type")]
+        public async Task AnonymousDelegateTaskOfTReturnsNull()
+        {
+            var test = @"
+using System.Threading.Tasks;
+
+class Test
+{
+    public void Foo()
+    {
+        Task.Run<object>(delegate
+        {
+            return [|null|];
```

Comment:
I am not sure how to find the type from such syntax. We could have an explicit or implicit type declared on the `Run` but we could also have this delegate part of a variable declaration `Func<Task<object>> f = ...` .

Reply:
The key was to call `SemanticModel.GetTypeInfo` on the null literal expression. The compiler has to know what type that converts to, even if the value is `null`. So you don't even need to know the return type per se. Just what the null is interpreted as.
I went ahead and revised your code fix to use the `SyntaxGenerator` as well so that it works for VB as your analyzer already did.